### ISSUE TITLE
Support for classes(), etc

### ISSUE THREAD

**Issue by tim-hutchinson** [vuejs/vue-test-utils#139]
This came up in #118. We've found it easier to work with tests that use the assertion library's comparison functions, rather than the comparison functions like `hasClass`, `hasProp`, etc. 

I'd like to propose adding helpers that are more like `html()` and `text()` that simply return a consumable form of the element feature requested. IE, `classes()` would return an array of class names on the element. I've got an implementation of `classes` and can submit a PR for it and others that make sense, but wanted to see if it made sense before going further.

**Comment by eddyerburgh** [comment]
I agree. Value assertions are more useful than boolean assertions.

We could replace hasClass, hasAttribute, and hasProp with `classes`, `attributes`, and `props`. 

We've got a good opportunity to remove these before the official release.

The one method I think is useful is hasStyle, because it allows you to check `hasStyle('color', 'white')` instead of returning `{color: 'rgb(255, 255, 255)}`

What do people think?

**Comment by lmiller1990** [comment]
I haven't had much occasion to use `attributes` or `props`, but I'd be for including them - `props` seems like it might be handy. `classes` is solved in #118 , and should probably be renamed `classes` in light of the updated functionality.

I agree that `hasStyle` should stay, I have used  it quite a lot (changing button colors, opacities, and the such). I don't see a `styles` helper being very useful.

**Comment by vouill** [comment]
I prefer personnaly prefer the notation `getClasses()` but other than that i like it too.

**Comment by eddyerburgh** [comment]
I also prefer the notation `getClasses`.

What do other people think?

**Comment by tim-hutchinson** [comment]
The `getX` syntax seems different from the conventions used in this library already (`html`, `text`). 

Alternatively, it could go fully the other way and implement the functions as getters, so they would expose a property-like API `expect(wrapper.classes).to.eql(['foo', 'bar'])`

**Comment by eddyerburgh** [comment]
I've changed my mind, as you said we already have html and text methods.

**Comment by eddyerburgh** [comment]
This has been released in 1.0.0-beta.6

Happy to continue discussing API, but thinking about it I think:

```js
wrapper.props().someProp
```

Is more elegant that

```js
wrapper.getProps().someProp
```

Plus we can extend the `props` method by adding an optional param:

```js
wrapper.props('someProp')
```
